movieApp/movieSeriesListViewController.swift

```diff
@@ -0,0 +1,118 @@
+        
+        collectionView = UICollectionView(frame: .zero, collectionViewLayout: layout)
+        collectionView.register(AllMoviesSectionCell.self, forCellWithReuseIdentifier: "MovieCollectionViewCell")
+        collectionView.delegate = self
+        collectionView.dataSource = self
+        collectionView.backgroundColor = .white
+        
+        view.addSubview(collectionView)
+        collectionView.autoPinEdgesToSuperviewEdges()
+        collectionView.reloadData()
+            
+    }
+    
+    
+    private func populateMoviesWithYears() {
+        for movie in movieUseCase.allMovies {
+            let movieData = MovieData(movieID: movie.id)
+            if let year = movieData.year {
+                moviesWithYears.append((movie, year))
+            }
+        }
+    }
+    
+    
+    func navigateToMovieDetails(with movieID: Int) {
+        let movieDetailsViewController = MovieDetailsViewController(movieID: movieID)
+        navigationController?.pushViewController(movieDetailsViewController, animated: true)
+    }
+}
+
+
+extension AllMoviesViewController: UICollectionViewDataSource {
+    func numberOfSections(in collectionView: UICollectionView) -> Int {
+        return 1
+    }
+
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return movieUseCase.allMovies.count
+    }
+
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "MovieCollectionViewCell", for: indexPath) as! AllMoviesSectionCell
+        let movieWithYear = moviesWithYears[indexPath.item]
+        cell.configure(with: movieWithYear.movie, year: movieWithYear.year, delegate: self) 
+        return cell
+    }
+}
+
+extension AllMoviesViewController: UICollectionViewDelegateFlowLayout {}
+
+
+
+
+
+
+
+
+
+
+
+
+
+//extension movieSeriesViewController: UICollectionViewDataSource {
+//    func numberOfSections(in collectionView: UICollectionView) -> Int {
+//        return 1 // Single section for all movies in the series
+//    }
+//
+//    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+//        return moviesWithYears.count  // Return filtered number of movies
+//    }
+//
+//    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+//        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "MovieCollectionViewCell", for: indexPath) as! AllMoviesSectionCell
+//        let movieWithYear = moviesWithYears[indexPath.item]
+//        cell.configure(with: movieWithYear.movie, year: movieWithYear.year) // CHANGE: Pass movie and year
+//        return cell
+//    }
+//}
+//
+//extension movieSeriesViewController: UICollectionViewDelegateFlowLayout {
+//    // Customize layout if needed
+//}
```

Comment:
No need to push commented out code. Please submit for review only code which is being used